### PR TITLE
Remove obsolete code generators

### PR DESCRIPTION
**Pull request by mahge**
@mahge
Remove the CodegenCSharp generator.
4481dc9

  - This generator is marked as obsolete.

@mahge
Remove the CodegenAdevs generator.
1f12dd2

  - This generator is marked as obsolete.

@mahge
Remove the CodegenJava generator.
aa68bb0

  - This generator is marked as obsolete.

@mahge
Remove the CodegenModelica generator.
034556c

  - This generator is marked as obsolete.

  - We could have probably left this one be for now considering that it
    is quite small and seems to be useful if used.

    However, it is already marked as obsolete.
    Plus we will forget it and end up never removing it.

@mahge
Remove the CodegenSparseFMI generator.
f5e0d1b

  - This generator was added by
    15a8310

    It has not had any (relevant) updates or fixes since then apart from
    making sure it compiles when other irrelevant changes were introduced
    to other components of the compiler.

  - If it turns out it is used by someone, you can just revert this commit.

### CODE REVIEW

File: OMCompiler/Compiler/Util/Flags.mo [hunk 1]
```diff
@@ -369,202 +369,200 @@ constant DebugFlag DISABLE_WINDOWS_PATH_CHECK_WARNING = DEBUG_FLAG(98, "disableW
...
+constant DebugFlag DUMP_DATARECONCILIATION = DEBUG_FLAG(192, "dataReconciliation", false,
   Gettext.gettext("Dumps all the dataReconciliation extraction algorithm procedure"));
-constant DebugFlag ARRAY_CONNECT = DEBUG_FLAG(194, "arrayConnect", false,
+constant DebugFlag ARRAY_CONNECT = DEBUG_FLAG(193, "arrayConnect", false,
   Gettext.gettext("Use experimental array connection handler."));
-constant DebugFlag COMBINE_SUBSCRIPTS = DEBUG_FLAG(195, "combineSubscripts", false,
+constant DebugFlag COMBINE_SUBSCRIPTS = DEBUG_FLAG(194, "combineSubscripts", false,
   Gettext.gettext("Move all subscripts to the end of component references."));
-constant DebugFlag ZMQ_LISTEN_TO_ALL = DEBUG_FLAG(196, "zmqDangerousAcceptConnectionsFromAnywhere", false,
+constant DebugFlag ZMQ_LISTEN_TO_ALL = DEBUG_FLAG(195, "zmqDangerousAcceptConnectionsFromAnywhere", false,
```
Comment (AnHeuermann) [inline OMCompiler/Compiler/Util/Flags.mo:565]: Do you have a script for this changes or did you do it all yourself?
I would like to have a script for this stuff

Comment (mahge) [reply OMCompiler/Compiler/Util/Flags.mo:565]: Nop. I had to do it manually.  (╯°□°）╯︵ ┻━┻

Comment (perost) [reply OMCompiler/Compiler/Util/Flags.mo:565]: In Vim, select the lines you want to change and press ctrl+a or ctrl+x depending on whether you want to increment or decrement all numbers :smile: I use it surprisingly often.

**Review by lochel — APPROVED** [review]
Good to strip the beast a bit 👍

File: OMCompiler/Compiler/Util/Flags.mo (hunk 1 above)
Comment (mahge) [reply OMCompiler/Compiler/Util/Flags.mo:565]: Ahh vim. That's outside my domain :)

Comment (AnHeuermann) [reply OMCompiler/Compiler/Util/Flags.mo:565]: Well if Vim can do it there should be a plugin for VS Code. I found this one: https://marketplace.visualstudio.com/items?itemName=nmsmith89.incrementor, but didn't test it yet.

Comment (mahge) [reply OMCompiler/Compiler/Util/Flags.mo:565]: I just tried it. It actually works. Time to remove some flags just for fun.

Comment (lochel) [reply OMCompiler/Compiler/Util/Flags.mo:565]: But it doesn't work on a selection 😞  just on the cursor position, right?

Comment (mahge) [reply OMCompiler/Compiler/Util/Flags.mo:565]: You can select multiple positions and then do the increment.  Alt+click -> increment

**Pull request merged**